packages/main/src/plugin/container-registry.ts

```diff
@@ -290,12 +291,24 @@ export class ContainerProviderRegistry {
       internalProvider.api = undefined;
       internalProvider.libpodApi = undefined;
 
+      // if is the timeout running dont start other for events
+      // connect ENOENT \.\pipe\machine_name at PipeConnectWrap.afterConnect [as oncomplete]
+      if (this.initializedConnectionAPI.get(containerProviderConnection.endpoint.socketPath)) {
+        return;
+      }
+
       // ok we had some errors so we need to reconnect the provider
       // delay the reconnection to avoid too many reconnections
       // retry in 5 seconds
-      setTimeout(() => {
-        this.setupConnectionAPI(internalProvider, containerProviderConnection);
-      }, this.retryDelayEvents);
+      this.initializedConnectionAPI.set(
+        containerProviderConnection.endpoint.socketPath,
+        setTimeout(() => {
+          this.initializedConnectionAPI.set(containerProviderConnection.endpoint.socketPath, false);
+          this.setupConnectionAPI(internalProvider, containerProviderConnection);
+        }, this.retryDelayEvents)
+          ? true
+          : false,
```

Comment:
This is non-sense ? `!!setTimeout(() => {}, 2000)` is always true, why checking the result of setTimeout ?